### PR TITLE
Add text label display function for notification area

### PR DESCRIPTION
**Pull request by zhuyaliang**
@raveit65  @yetist 
The notification area before modification can only display icons. This PR adds the text label display function.

The ```app_indicator_set_label```  API doesn't work before modification. After modification, the label can be displayed.

[Test code](https://github.com/mate-desktop/mate-panel/files/5891282/simple-test.tar.gz)

#### Use of test code:
``` 
mkdir /tmp/test-dir
cp -rf simple-test.tar.gz  /tmp/test-dir/
cd  /tmp/test-dir/
tar -xvf simple-test.tar.gz
make
./test
````
After running, the icon and text will be displayed in the notification area
2021-02-20 14-55-44屏幕截图

### CODE REVIEW

File: applets/notification_area/status-notifier/sn-item-v0.c
Comment (rbuj) [inline applets/notification_area/status-notifier/sn-item-v0.c:1416]: This change is not related to the commit title.

File: applets/notification_area/status-notifier/sn-item-v0.c
Comment (rbuj) [inline applets/notification_area/status-notifier/sn-item-v0.c:1131]: Please, fix the indentation.

File: applets/notification_area/status-notifier/sn-item-v0.c
Comment (rbuj) [inline applets/notification_area/status-notifier/sn-item-v0.c:333]: I don't think it should be called unconditionally.

File: applets/notification_area/status-notifier/sn-item-v0.c
Comment (rbuj) [inline applets/notification_area/status-notifier/sn-item-v0.c:912]: please, fix the indentation.

File: applets/notification_area/status-notifier/sn-item-v0.c
Comment (rbuj) [inline applets/notification_area/status-notifier/sn-item-v0.c:974]: please, fix the indentation.

File: applets/notification_area/status-notifier/sn-item-v0.c
Comment (zhuyaliang) [reply applets/notification_area/status-notifier/sn-item-v0.c:912]: @rbuj  To tell you the truth, I'm really bothered by the indent format of mate。It takes more time to modify indentation than to modify code，Is the indent here two spaces？

**Review by rbuj — COMMENTED** [review]
Anyway I don't think a label should be shown on the icons of the notification area.
https://specifications.freedesktop.org/systemtray-spec/systemtray-spec-0.3.html

File: applets/notification_area/status-notifier/sn-item-v0.c
Comment (rbuj) [inline applets/notification_area/status-notifier/sn-item-v0.c:62]: please, fix the indentation.

File: applets/notification_area/status-notifier/sn-item-v0.c
Comment (rbuj) [reply applets/notification_area/status-notifier/sn-item-v0.c:912]: Where did you see a single blank space being used for indentation?

File: applets/notification_area/status-notifier/sn-item-v0.c
Comment (zhuyaliang) [reply applets/notification_area/status-notifier/sn-item-v0.c]: I think it's unconditional

**Comment by zhuyaliang** [comment]
> Anyway I don't think a label should be shown on the notification icons.
I think it's a very practical function

File: applets/notification_area/status-notifier/sn-item-v0.c
Comment (rbuj) [inline applets/notification_area/status-notifier/sn-item-v0.c:862]: missing blank line?

File: applets/notification_area/status-notifier/sn-item-v0.c
Comment (raveit65) [reply applets/notification_area/status-notifier/sn-item-v0.c]: > To tell you the truth, I'm really bothered by the indent format of mate。

And i am really bothered to mark always the same formatting problems by people.

**Comment by zhuyaliang** [comment]
@raveit65 What do you think of this new function

File: applets/notification_area/status-notifier/sn-item-v0.c
Comment (rbuj) [inline applets/notification_area/status-notifier/sn-item-v0.c:859]: memory leak

**Review by lukefromdc — COMMENTED** [review]
Status notfier icons didn't show up with this change on my system, and about half the time I got a segfault on restarting the panel

File: applets/notification_area/status-notifier/sn-item-v0.c
Comment (zhuyaliang) [reply applets/notification_area/status-notifier/sn-item-v0.c]: @rbuj 
If not changed, the status icon will not be displayed correctly after restarting the panel

**Comment by zhuyaliang** [comment]
@lukefromdc Please try again to see if the status icon can be displayed normally，I restart the panel 10 times in a row, and there is no segment error

**Review by lukefromdc — APPROVED** [review]
This now works perfectly, and the file even builds with no warnings whatsoever. The test code built fine and when run showed a beautiful little icon plus changing text, showing this works as intended.  I like it, and `since app_indicator_set_label` is part of the specification we should make it work

Distributing the test code as source with a makefile was a nice touch-and in my case a requirement as it allowed me to examine the code and ensure nobody submitted a PR whose real job was to get a rootkit onto my box in the process of testing it. A test blob on someone's VM used for nothing but testing PR's would not be very dangerous but I am testing on a live system, on bare metal.

**Comment by rbuj** [comment]
The code of the example is quite similar to the link below, the PR author removed the copyright header and license info...

- https://github.com/AyatanaIndicators/libayatana-appindicator/tree/master/example

Please, do not remove the copyright notice and include the info about where the code comes from.

**Comment by rbuj** [comment]
The signal "XAyatanaNewLabel" was reported on https://github.com/mate-desktop/mate-panel/issues/571 and the segfault was superseded by a debug message on https://github.com/mate-desktop/mate-panel/commit/862607a51aade39bdb65f266cdd7b23b3e0eca89.
I don't want to block it if the other members of the team believe that this signal shouldn't be ignored from now on.
Having said that, @sunweaver @flexiondotorg are the suitable people to comment on this topic.

**Comment by raveit65** [comment]
> The signal "XAyatanaNewLabel" was reported on #571 and the segfault was superseded by a debug message on 862607a.

Out of question, a segfault should be fixed. I don't like to receive bug reports.

**Comment by lukefromdc** [comment]
According to that, the segfault was worked around, but leaving us with a debug message that should still be fixed. The segfault I got with an early version of this PR was fixed with the update 
https://github.com/mate-desktop/mate-panel/pull/1197/commits/91239350ff77abbe7b396c7cd1d57ad96087268e
which worked perfectly in my test

**Comment by lukefromdc** [comment]
Any updates on this? Should we merge this, wait for another review, or what?

**Comment by zhuyaliang** [comment]
@yetist @raveit65 Why didn't anyone discuss this PR

**Comment by lukefromdc** [comment]
There are a LOT of open PR's right now, so no surprise it takes people a while to get back to this now-older one.
Right now we have one vote for and one vote against this. The PR works, so the problem is not having to wait for technical fixes. Rather we need to decide as a team what to do with this, do we want this new feature that the indicator (as opposed to the older tray) specification does call for? Merge it, close it, or something else?

**Comment by raveit65** [comment]
> Why didn't anyone discuss this PR

I will give the question back. Why you never helped out with reviews for mate-control-center, as maintainer of m-c-c?
This would help to reduce the long review queue.

**Comment by raveit65** [comment]
Your simple-test.tar.gz is corrupt.

**Comment by raveit65** [comment]
I have no idea how to test your PR....

**Comment by raveit65** [comment]
Looks like engrampa has an issue to extract your tar.gz, but i was able to extract with Tar itself-
But what should i do with simple-client.c ?
Better write a description before you ask for more reviewer.....

**Comment by rbuj** [comment]
> Your simple-test.tar.gz is corrupt.

It's because of `simple-test.tar.gz` isn't a compressed archive, it's only an archive, ie  it's an uncompressed tar archive. 😄 
```
$ file simple-test.tar.gz 
simple-test.tar.gz: POSIX tar archive (GNU)
$ tar -tvf simple-test.tar.gz 
-rw-r--r-- mate/mate      1973 2021-01-29 03:44 simple-client.c
-rw-r--r-- mate/mate       191 2021-01-29 03:46 Makefile
```

**Comment by zhuyaliang** [comment]
> I will give the question back. Why you never helped out with reviews for mate-control-center, as maintainer of m-c-c?
> This would help to reduce the long review queue.

Sorry, this is my personal problem. I admit this mistake. I will participate in the audit of mate-c-c as soon as possible

**Comment by raveit65** [comment]
Hmm, this doesn't work here.
Mate-panel is compiled with your commit but i see only the icon and clicking on"Hide/show-Label" in menu doesn't have any effect.
And SNI Support is enabled.
```
[rave@mother ~]$ gsettings get org.mate.panel enable-sni-support
true
```

**Comment by zhuyaliang** [comment]
> Hmm, this doesn't work here.

Sorry, I didn't make it clear. We need to reload the ```notification-area-applet```
Using the```pkill -f notification-area-applet``` command and Click reload

**Comment by raveit65** [comment]
I did another mistake, sorry my fault, but now i got it running. I do not see any warning so it seems OK for me.

> According to that, the segfault was worked around, but leaving us with a debug message that should still be fixed. 

@lukefromdc 
From what debug message are you talking? In build log?

**Comment by rbuj** [comment]
@raveit65 without this pull request there is no segfault because @monsta replaced `g_assert_not_reached` with `g_debug`:
 
1. remove the notification area applet from the panel
2. launch the command below from a terminal
```
$ G_MESSAGES_DEBUG=all /usr/libexec/notification-area-applet 
```
3. add a notification area applet to a panel
4. run the test application provided in PR comment (tar xvf .. make)
```
$ ./test
```
5. A debug message is already shown (step 2):
```
<cut>
(notification-area-applet:36584): notification-area-applet-DEBUG: 13:30:36.162: signal 'XAyatanaNewLabel' not handled!
```

**Comment by raveit65** [comment]
Ok, i need to rebuild the panel with out-of-process-applets.

**Comment by lukefromdc** [comment]
Not at my desktop at the moment, will have to look that up later

**Comment by raveit65** [comment]
I see other warnings of this Ayatana code by debian people, but i don't get this one.
```
[rave@mother ~]$ G_MESSAGES_DEBUG=all /usr/libexec/mate-panel/notification-area-applet 
<cut>

(notification-area-applet:107395): notification-area-applet-DEBUG: 16:14:42.706: property 'XAyatanaLabelGuide' not handled!
(notification-area-applet:107395): notification-area-applet-DEBUG: 16:14:42.706: property 'XAyatanaOrderingIndex' not handled!
[rave@mother ~]$ 

Honestly, i don't care about if there is one warning more, and i don't get them because my goal is to build the panel with in-process-applets to support wayland and X11.
So, we can merge it, ihmo.

**Review by raveit65 — APPROVED** [review]
LGTM,
builds and runs fine.

File: applets/notification_area/status-notifier/sn-item-v0.c
Comment (muktupavels) [inline applets/notification_area/status-notifier/sn-item-v0.c:859]: This returns newly allocated string! You have memory leak here...

Comment (rbuj) [reply applets/notification_area/status-notifier/sn-item-v0.c:859]: PR #1204

**Pull request merged**
